scripts/listapranzo.js

```diff
@@ -114,11 +149,58 @@ module.exports = function (robot) {
     msg.send('Se ordinate al TuttoBene posso aiutarvi io!');
   });
 
+  robot.respond(/develunch([\s\S]*)?/i, function (msg) {
+    var when = (msg.match[1] || '').trim();
+
+    if (when === '') {
+      var develunch = robot.brain.get('develunch') || null;
+      if (develunch === null) {
+        msg.reply('Develunch non impostato!');
+      } else {
+        msg.reply('Il develunch sarà ' + develunch.toDateString());
+      }
+      return;
+    }
+
+    var this_week = /(quest|oggi|domani|corrente|attual)/i.test(when);
+    var next_week = /(prossim|successiv|seguent)/i.test(when);
+    var now = new Date();
+    var friday_offset = 5 - now.getDay();
+
+    if (this_week && !next_week) {
+      friday_offset += 0;
+    } else if (next_week && !this_week) {
+      friday_offset += 7;
+    } else {
+      msg.reply('non riesco a capire quando sia il develunch... prova a dirlo in un altro modo!');  
+      return;
+    }
+
+    var next_develunch = now;
+    /* setDate() tiene conto anche del cambio mese/anno, anche se friday_offset è >31 o <0 */
+    next_develunch.setDate(now.getDate() + friday_offset);
+
+    msg.reply('Ok, develunch impostato per ' + next_develunch.toDateString());
+    robot.brain.set('develunch', next_develunch);
+  });
+
+  var isDevelunch = function() {
+    var tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+
+    return isToday(robot.brain.get('develunch') || tomorrow);
+  };
+
   robot.respond(/per me (.*)/i, function (msg) {
     var dish = msg.match[1].trim();
     var user = msg.message.user;
     var order = getOrder(msg);
 
+    if (isDevelunch()) {
```

Comment:
si può spostare più in alto, come prima istruzione per evitare di fare lavoro inutile